tests/unit_tests/blas/batch/gemm_batch_stride.cpp

```diff
@@ -48,12 +48,65 @@ extern std::vector<sycl::device *> devices;
 namespace {
 
 template <typename fp>
+typename std::enable_if<std::is_integral<fp>::value, bool>::type check_equal_int(fp x, fp x_ref,
+                                                                                 int error_mag) {
+    return (std::abs(x - x_ref) <= 1);
+}
+
+template <class T>
+struct acc_type {
+    typedef host_accessor<T, 1, access_mode::read> type;
+};
+
+template <class T>
+using acc_type_t = typename acc_type<T>::type;
+
+template <class T>
+struct vec_type {
+    typedef vector<T, allocator_helper<T, 64>> type;
+};
+
+template <class T>
+using vec_type_t = typename vec_type<T>::type;
+
+// Check for int32_t and Ts=float as small differences in the reference become large after rounding
+bool check_equal_matrix_int(acc_type_t<int32_t> &M, vec_type_t<int32_t> &M_ref,
+                            oneapi::mkl::layout layout, int m, int n, int ld, int error_mag,
+                            std::ostream &out) {
+    bool good = true;
+    int idx, count = 0;
+    for (int j = 0; j < n; j++) {
+        for (int i = 0; i < m; i++) {
+            idx = (layout == oneapi::mkl::layout::col_major) ? i + j * ld : j + i * ld;
+            if (!check_equal_int(M[idx], M_ref[idx], error_mag)) {
+                out << "Difference in entry (" << i << ',' << j << "): DPC++ " << M[idx]
+                    << " vs. Reference " << M_ref[idx] << std::endl;
+                good = false;
+                count++;
+                if (count > MAX_NUM_PRINT)
+                    return good;
+            }
+        }
+    }
+
+    return good;
+}
+
+template <typename T>
+bool check_mat(acc_type_t<T> &M, vec_type_t<T> &M_ref, oneapi::mkl::layout layout, int m, int n,
+               int ld, int error_mag, std::ostream &out) {
+    if constexpr (std::is_same<T, int32_t>::value)
+        return check_equal_matrix_int(M, M_ref, layout, m, n, ld, error_mag, out);
+    return check_equal_matrix<acc_type_t<T>>(M, M_ref, layout, m, n, ld, error_mag, out);
```

Comment:
Thanks, I'm running the CI to make sure this is not giving us any warning. It's worth adding a comment to explain the issues with specializing `check_equal_matrix`.

Reply:
This has been refactored so the comment would not make much sense anymore